alf/algorithms/hypernetwork_algorithm_test.py

```diff
@@ -197,6 +194,189 @@ def _test(i, sampled_predictive=False):
         self.assertLess(mean_err, 0.5)
         self.assertLess(cov_err, 0.5)
 
+    @parameterized.parameters((True), (False, False, False, 48), (False, True),
+                              (True, False, True))
```

Comment:
I performed one more test that I think is crucial.  (direct_jac=False, block_inverse_mvp=True, lambda_trainable=True)
This performed nearly on-par with (direct_jac=True, block_inverse_mvp=False, lambda_trainable=True), that is to say, quite well.